src/Screens/Authenticate/Authenticate.js

```diff
@@ -0,0 +1,142 @@
+import "./authenticate.css";
+import { useNavigate } from "react-router-dom";
+import { useRef } from "react";
+import { useToast } from "./../../Context/toast-context";
+import { useNavbar } from "./../../Context/navbar-context";
+import { useToken } from "./../../Context/token-context";
+
+function Authenticate(props) {
+  const { setEncodedToken } = useToken();
+
+  let navigate = useNavigate();
+
+  const { toggleToast, toastVisibility, toastColor, toastText } = useToast();
+
+  const { setNavbarButtonText } = useNavbar();
+
+  const emailPattern = /\S+@\S+\.\S+/;
+  const emailRef = useRef(null);
+  const passwordRef = useRef(null);
+
+  const handleAuth = () => {
+    var payload = {
+      email: emailRef.current.value,
+      password: passwordRef.current.value,
+    };
+
+    if (props.cardTitle === "LOGIN") {
+      fetch("/api/auth/login", {
```

Comment:
fetch is fine but try using axios. it is more convinient and easier to work with. Also it makes code look short and clean.